srtcore/channel.cpp

```diff
@@ -90,86 +88,92 @@ m_iIpV6Only(-1)
 {
 }
 
-CChannel::CChannel(int version):
-m_iIPversion(version),
-m_iSocket(),
-#ifdef SRT_ENABLE_IPOPTS
-m_iIpTTL(-1),
-m_iIpToS(-1),
-#endif
-m_iSndBufSize(65536),
-m_iRcvBufSize(65536),
-m_iIpV6Only(-1),
-m_BindAddr(version)
+CChannel::~CChannel()
 {
-   SRT_ASSERT(version == AF_INET || version == AF_INET6);
-   m_iSockAddrSize = (AF_INET == m_iIPversion) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6);
 }
 
-CChannel::~CChannel()
+void CChannel::createSocket(int family)
 {
+    // construct an socket
+    m_iSocket = ::socket(family, SOCK_DGRAM, IPPROTO_UDP);
+
+#ifdef _WIN32
+    int invalid = INVALID_SOCKET;
+#else
+    int invalid = -1;
+#endif
+
+    if (m_iSocket == invalid)
```

Comment:
Use `const`. It might be better to define `INVALID_SOCKET` for platforms other than Windows.

Reply:
Platforms other than Windows follow the POSIX definition and use -1. At least those that this project supports.